pkg/controller/bundlec/bundle_sync_task.go

```diff
@@ -246,7 +246,11 @@ func (st *bundleSyncTask) findObjectsToDelete() error {
 			gvk = res.Spec.Object.GetObjectKind().GroupVersionKind()
 			name = res.Spec.Object.(meta_v1.Object).GetName()
 		} else if res.Spec.Plugin != nil {
-			gvk = st.pluginContainers[res.Spec.Plugin.Name].Plugin.Describe().GVK
+			plugin, ok := st.pluginContainers[res.Spec.Plugin.Name]
+			if !ok {
+				return errors.Errorf("plugin %q is not a valid plugin", res.Spec.Plugin.Name)
```

Comment:
This should not break the loop - see below. This error should be ignored because it must have been reported earlier.

Reply:
Need a similar comment as below then, justifying a silent `continue`

Reply:
I was not right, `continue` is not the right behaviour here. This loops filters out things that should not be deleted. So if we continue, we might delete a thing that should not be deleted. Thanks @yiyangc91 for catching this.